### DISCUSSION TITLE
Properties, members, items, and elements

### DISCUSSION THREAD

**Discussion by elreydetodo** [json-schema-org/community#44] [Vision and direction]
[RFC 8259](https://datatracker.ietf.org/doc/html/rfc8259) defines "The JavaScript Object Notation (JSON) Data Interchange Format". It is _the_ definition of JSON.

Here's the beginning of section 4:
```
4.  Objects

   An object structure is represented as a pair of curly brackets
   surrounding zero or more name/value pairs (or members).
...
```

Here's the beginning of section 5:
```
5.  Arrays

   An array structure is represented as square brackets surrounding zero
   or more values (or elements).
...
````

The actual RFC for JSON calls the things that JSON Schema's "property" a "member", and JSON Schema's "item" an "element".

[JSON Schema Core Draft 2020-12](http://json-schema.org/draft/2020-12/json-schema-core.html) says this in section 2:
```
The terms "JSON", "JSON text", "JSON value", "member", "element", "object", "array", "number", "string", "boolean", "true", "false", and "null" in this document are to be interpreted as defined in RFC 8259.
```

We even _purport_ to use the vocabulary from the RFC, then we _just don't do it_.

JSON Schema is definitely a lower precedence specification than JSON itself. It seems like JSON Schema should be using the terminology established by the RFC instead of inventing new terms.

Am I missing something about why we are using terminology that is not consistent with the JSON RFC? Do we have any plan to change this? How did we get to this point?

**Comment by jdesrosiers** [comment]
You bring up a good point. The terminology used in JSON Schema pre-dates anyone currently working on the spec, so no one knows exactly how the discrepancy came to be. The terminology appears to come from JavaScript. I don't know why JSON chose to use different terminology than JavaScript despite being born form JavaScript.

I think changing the terminology used in the spec to match the RFC makes sense, but I wouldn't change the keywords to match JSON terminology. JSON Schema is too widely used to make such drastic changes to the terminology. We're stuck with it at this point.
